src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -1219,6 +1252,65 @@ internal async void OnTimer(object sender, System.Timers.ElapsedEventArgs e)
             }
         }
 
+        internal void OnInvocationTimer(object sender, System.Timers.ElapsedEventArgs e)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            try
+            {
+                CheckAndRecycleFaultyLanguageWorker();
+            }
+            catch
+            {
+                // Don't allow background execptions to escape
+                // E.g. when a rpc channel is shutting down we can process exceptions
+            }
+            try
+            {
+                _invocationTimer.Start();
+            }
+            catch (ObjectDisposedException)
+            {
+                // Specifically ignore this race - we're exiting and that's okay
+            }
+        }
+
+        internal void CheckAndRecycleFaultyLanguageWorker()
+        {
+            _workerChannelLogger.LogInformation("Checking and recycling faulty language workers");
+
+            Dictionary<string, int> failureRatePerWorkerId = new Dictionary<string, int>();
+            var failureThreshold = 30; // ToDo: Moving this constant once finalized
```

Comment:
@mathewc thoughts on `failureThreshold` value? Right now, this code will just **log** the workerIds that are identified as faulty workers to be recycled. Does `30` look good to help us examine Prod apps?